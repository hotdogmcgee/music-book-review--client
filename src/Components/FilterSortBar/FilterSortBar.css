.filter-sort-buttons {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    display: flex;
    justify-content: space-evenly;
    flex-wrap: wrap;
}

.red {
    background-color: red
}

.options-list {
    list-style: none;
    padding: 0 20vw;
    display: flex;
    justify-content: space-evenly;
    flex-wrap: wrap;
}

.option-list-item {
    display: flex;
    justify-content: space-around;
    border: 1px solid black;
    padding: 5px;
    text-align: center;
    width: 45%;
    margin: 5px;
    cursor: pointer;
}

.filter-highlight {
    background-color: red
}

i {
    border: solid black;
    border-width: 0 3px 3px 0;
    display: inline-block;
    padding: 3px;
  }

  .up {
    transform: rotate(-135deg);
    -webkit-transform: rotate(-135deg);
  }

  .down {
    transform: rotate(45deg);
    -webkit-transform: rotate(45deg);
  }
