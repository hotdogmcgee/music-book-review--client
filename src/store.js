export const STORE = {
    bookList: [
        {
            instrument: 'piano',
            image: 'an image',
            title: 'where the red fern grows',
            author: 'zi forgot',
            description: 'two dogs in a very sad tale',
            rating: 4,
            numReviews: 10,
            id: 1,
            published_year: 2001,
            cost: 30,
            reviews: [
                {
                    user: {
                        full_name: 'dave bones'
                    },
                    rating: 3,
                    id: 1,
                    text: 'it was pretty good'
                }
            ]
        },
        {
            instrument: 'guitar',
            image: 'another image',
            title: 'Harry Potter and the Goblet of Fire',
            author: 'JK Rowling',
            description: 'everyone knows it by now',
            rating: 3,
            numReviews: 3,
            cost: 20,
            published_year: 1990,
            id: 2
        }
    ]
}