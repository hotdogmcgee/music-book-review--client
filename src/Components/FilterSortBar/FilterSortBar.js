import React from "react";
import { Section } from "../Utils/Utils";
import FilterButton from "./FilterButton/FilterButton";
import SortButton from "./SortButton/SortButton";
import "./FilterSortBar.css";

export default class FilterSortBar extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      displayOptions: false,
      list: [],
      filters: ["under 25 dollars", "old books", "recent"],
      filterOrSort: null,
      activeFilters: {
        "under 25 dollars": true,
        "old books": true,
        'recent': true
      }
    };

    this.handleSetOptions = this.handleSetOptions.bind(this);
    this.handleSortOptionClick = this.handleSortOptionClick.bind(this);
  }

  toggleShowOptions = () => {
    this.setState({
      displayOptions: !this.state.displayOptions
    })
  };

  setFilterOrSort = value => {
    this.setState({
      filterOrSort: value
    });
  };

  toggleClass = value => {
    const activeFilters = { ...this.state.activeFilters };
    activeFilters[value] = !this.state.activeFilters[value];
    this.setState({ activeFilters });
  };

  handleSetOptions(list) {
    this.setState({
      list
    });
  }

  handleOptionClick(value) {
    if (this.state.filterOrSort === "filter") {
      this.toggleClass(value);
      this.handleFilterOptionClick(value);
    } else if (this.state.filterOrSort === "sort") {
      this.handleSortOptionClick(value);
    } else {
      return;
    }
  }
  // handleFilterOptionClick(filterValue) {
  //   this.props.onFilterOptionClick(filterValue);
  // }

  handleFilterOptionClick(value) {
    const filters = this.state.filters;
    let newFilters;
    if (filters.includes(value)) {
      debugger;

      newFilters = filters.filter(item => item !== value);
      this.setState({
        filters: newFilters
      });
    } else {
      filters.push(value);
      newFilters = filters;
      this.setState({
        filters: newFilters
      });
    }

    this.props.onFilterOptionClick(newFilters);
  }

  handleSortOptionClick(sortValue) {
    this.props.onSortOptionClick(sortValue);
  }
  renderOptionsList(list = []) {
    list = this.state.list;

    return (
      <ul className="options-list">
        {list.map((item, index) => {
          return (
            <li
              className={`option-list-item ${
                this.state.activeFilters[item]
                  ? "filter-highlight"
                  : "no-filter-highlight"
              }`}
              key={index}
              onClick={() => this.handleOptionClick(item)}
            >
              {item}
            </li>
          );
        })}
      </ul>
    );
  }

  render() {
    return (
      <Section id="filter-sort-bar">
        <div className="filter-sort-buttons">
          <FilterButton
            toggleShowOptions={this.toggleShowOptions}
            setOptions={this.handleSetOptions}
            toggleFilterOrSort={this.setFilterOrSort}
          />
          <SortButton
            toggleShowOptions={this.toggleShowOptions}
            setOptions={this.handleSetOptions}
            toggleFilterOrSort={this.setFilterOrSort}
          />
        </div>
        <div>{this.state.displayOptions ? this.renderOptionsList() : null}</div>
      </Section>
    );
  }
}
