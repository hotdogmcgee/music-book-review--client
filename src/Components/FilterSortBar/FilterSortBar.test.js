import React from 'react';
import FilterSortBar from './FilterSortBar.js'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { STORE } from '../../store.js'

describe('FilterSortBar component', () => {

      it('renders a FilterSortBar by default', () => {
        const wrapper = shallow(<FilterSortBar />)
        expect(toJson(wrapper)).toMatchSnapshot()
      })
    

})