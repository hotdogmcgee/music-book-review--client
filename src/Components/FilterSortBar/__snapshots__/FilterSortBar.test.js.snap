// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterSortBar component renders a FilterSortBar by default 1`] = `
<Section
  id="filter-sort-bar"
>
  <div
    className="filter-sort-buttons"
  >
    <SortButton
      setOptions={[Function]}
      showOptions={[Function]}
      toggleFilterOrSort={[Function]}
      toggleShowOptions={[Function]}
    />
  </div>
  <div />
</Section>
`;
